### DISCUSSION TITLE
Facing Module not found issue with Recharts: Error: Package path ./helpers/interopRequireDefault?commonjs-external is not exported from package.

### DISCUSSION THREAD

**Discussion by chiraagByjus** [recharts/recharts#3509] [General]
I am facing a issue with recharts library. I have a component library which I have published on github packages using Rollup as my bundler. The component library is developed in reactjs. So when I install this library on a consumer application(reactjs web app), I'm getting the error as shown in the attached screenshot. Until a month and a half ago everything was working fine. After which I published a new version and ever since that I'm facing the same issue. The version previous to this new version that I published a month and a half ago is working fine but anything published after this version throws up the same error. If I remove recharts dependency from the library and the component using recharts, I am able to consume the library with no issues. Not sure how to resolve this issue. Request recharts community help to resolve this issue. 

<img width="1066" alt="Screenshot 2023-04-10 at 12 40 33 PM" src="https://user-images.githubusercontent.com/109967209/231478872-63d1fc2c-a560-4c80-add0-bf422d60358e.png">

**Comment by ckifer** [comment]
Hey, sorry you're experiencing this!

It is probably similar to this https://github.com/recharts/recharts/issues/3448 

And most likely due to our use of victory-vendor. The error you are receiving doesn't make much sense though...

**Comment by chiraagByjus** [comment]
@ckifer So I checked the above #3448 discussion. And I checked my library codebase, so I am using Recharts 2.2.0 version, and victory-vendor package is not installed on the library itself. I wasn't able to find it in the node_modules either. However, if I install the latest version of Recharts, then victory-vendor also gets added to node_modules. But on either scenario, the error I'm facing is the same as the error seen in the previously attached screenshot.

Both my library & consumer applications are bootstrapped by CRA. Ejecting on either of them is not an option for me. So as suggested in the #3448 thread, I won't be able to change webpack configuration.

**Reply by ckifer** [reply]
Well good to know that it happens pre-victory vendor then. 

What is the last version of recharts that works for you? Any way you can keep decreasing version until it works to isolate where the breaking change happened?

**Reply by chiraagByjus** [reply]
Let me try this.

**Reply by ckifer** [reply]
Looks babel related. So I'm thinking a combo of babel config changes on our end + roll up on your end might be causing the break. The error message isn't very helpful though 😅

**Reply by chiraagByjus** [reply]
@ckifer I tried downgrading the recharts version but no luck. So the thing is I have been using recharts with my library since december. In my package.json, in the dependencies, I had ` recharts:"^2.2.0"`. This was working fine, I had published my library and was able to consume it as well. In mid of february, I updated some other components, didn't touch rehcarts version or the components related to it but since then, I'm facing this interopRequest issue. 

Here's the contents of my babel.rc config: 
<img width="777" alt="Screenshot 2023-04-14 at 11 10 19 AM" src="https://user-images.githubusercontent.com/109967209/231951212-7e23c756-461d-49fb-ad49-160f11776915.png">

The contents of my rollup config: 
<img width="992" alt="Screenshot 2023-04-14 at 11 14 31 AM" src="https://user-images.githubusercontent.com/109967209/231951928-845b0a0a-ed8f-41ad-892a-509a384bca0e.png">
<img width="989" alt="Screenshot 2023-04-14 at 11 14 44 AM" src="https://user-images.githubusercontent.com/109967209/231951949-e29ea932-7d3b-477b-8d82-51aa5a43d321.png">


Package.json file: 
{
  "name": "@chiraagbyjus/trial-library-dummy",
  "version": "0.0.16",
  "description": "",
  "main": "dist/index.cjs.js",
  "scripts": {
    "storybook": "start-storybook -p 6006",
    "build-storybook": "build-storybook",
    "build:rollup": "rollup -c --bundleConfigAsCjs",
    "test": "react-scripts test"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/chiraagByjus/trial-library-dummy.git"
  },
  "keywords": [],
  "author": "",
  "license": "ISC",
  "devDependencies": {
    "@babel/cli": "^7.19.3",
    "@babel/core": "^7.19.6",
    "@babel/plugin-transform-runtime": "^7.19.6",
    "@babel/preset-env": "^7.19.4",
    "@babel/preset-react": "^7.18.6",
    "@peculiar/webcrypto": "^1.4.1",
    "@rollup/plugin-babel": "^6.0.2",
    "@rollup/plugin-commonjs": "^24.0.0",
    "@rollup/plugin-image": "^3.0.1",
    "@storybook/addon-actions": "^6.5.13",
    "@storybook/addon-essentials": "^6.5.13",
    "@storybook/addon-interactions": "^6.5.13",
    "@storybook/addon-links": "^6.5.13",
    "@storybook/builder-webpack4": "^6.5.13",
    "@storybook/manager-webpack4": "^6.5.13",
    "@storybook/preset-scss": "^1.0.3",
    "@storybook/react": "^6.5.13",
    "@storybook/testing-library": "^0.0.13",
    "@testing-library/jest-dom": "5.16.5",
    "@testing-library/react": "13.4.0",
    "@testing-library/user-event": "13.5.0",
    "@types/dompurify": "^2.4.0",
    "autoprefixer": "^10.4.13",
    "babel-loader": "^8.2.5",
    "babel-plugin-module-resolver": "^5.0.0",
    "babel-plugin-recharts": "^2.0.0",
    "css-loader": "^3.6.0",
    "postcss": "^8.4.18",
    "postcss-loader": "^6.2.1",
    "prop-types": "^15.8.1",
    "react": "18.2.0",
    "react-dom": "18.2.0",
    "react-scripts": "5.0.1",
    "rollup": "^3.2.5",
    "rollup-plugin-sourcemaps": "^0.6.3",
    "rollup-plugin-styles": "^4.0.0",
    "sass": "^1.56.0",
    "sass-loader": "^10.4.1",
    "storybook-addon-sass-postcss": "^0.1.3",
    "style-loader": "^1.3.0"
  },
  "peerDependencies": {
    "react": "^18.2.0",
    "react-dom": "^18.2.0"
  },
  "dependencies": {
    "@babel/runtime": "^7.20.1",
    "@rollup/plugin-node-resolve": "^15.0.1",
    "shaka-player": "^4.2.3",
    "recharts": "2.1.6"
  },
  "publishConfig": {
    "registry": "https://npm.pkg.github.com"
  }
}

Request you to let me know if I have to make any changes.

**Reply by ckifer** [reply]
Interesting. So it seems recharts isn't the issue then?

I'm glad to help you debug if I can, but just want to make sure it wasn't something we did in the latest versions

**Reply by ckifer** [reply]
The only way you'll be able to determine what this is is if you find the diff that caused the initial failure and compare against your current code. I can't really be of much assistance since it isn't a recharts issue.

**Reply by chiraagByjus** [reply]
@ckifer So the issue is, when I include recharts as a dependency on my library and make use of it, thats when I face this issue. If I don't use recharts, then it works fine. I am able to consume it on other applications. Isn't this due to recharts then? I installed recharts ^2.2.0 version on my library in December and it was working fine until the version that I published in mid January and I was able to consume it as well. Then mid/end feb I tried to publish another version, it got published but I was not able to consume it and since then I'm facing this above issue. 

I have also reverted all the code back to the last version that was working fine and still I'm facing this issue. I don't know if this is something to do with babel version being incompatible with recharts. I tried changing the babel version to the version in recharts package.json as well, but that also didnt work.

**Reply by ckifer** [reply]
No changes were backported to any old versions of recharts. Since you have installed the previous version that was working and it no longer works this means that the breaking change does not lie with recharts but most likely babel (or another transitive dep).

Recharts is compiled with babel, the config is [here](https://github.com/recharts/recharts/blob/master/.babelrc#L1-L37). I would check for changes to your babel version in your lockfile and try reverting that change. If you can trace down the bad dependency that causes this you may have an easier path towards resolution

**Reply by chiraagByjus** [reply]
Understood. Okay let me have a look at this, I'll try downgrading babel version. Thanks for the support.

**Reply by chiraagByjus** [reply]
Hey @ckifer , just wanted to know if circular dependencies would cause this issue. So I get the warning as shown in the aatached screenshot, when I use rollup to bundle the  library:
<img width="1084" alt="Screenshot 2023-04-21 at 1 49 24 PM" src="https://user-images.githubusercontent.com/109967209/233583332-e1c7e9ec-a982-4468-aab4-54967ed67e6c.png">

I tried using the specific version of babel packages both on my react consumer app & my library which were published 3 months ago, the last time I was able to publish and consume my library. I still have not been able to resolve it. I have reached out to babel team via discussions on their github profile, but no response yet. Any suggestions?

**Reply by chiraagByjus** [reply]
@ckifer I've uploaded the library codebase on Github: https://github.com/chiraag918/trial-library-dummy. Please, feel free to play around with it to check if recharts has any issues.

**Reply by ckifer** [reply]
The best thing to do would be to go through your lock file differences when your issue occurred and check everything that changed
